src/sources/docker_logs.rs

```diff
@@ -75,26 +77,29 @@ impl Default for DockerLogsConfig {
 }
 
 impl DockerLogsConfig {
-    fn container_name_included<'a>(
+    fn container_name_or_id_included<'a>(
         &self,
         id: &str,
         names: impl IntoIterator<Item = &'a str>,
     ) -> bool {
-        if let Some(include_containers) = &self.include_containers {
-            let id_flag = include_containers
-                .iter()
-                .any(|include| id.starts_with(include));
+        let containers: Vec<String> = names.into_iter().map(Into::into).collect();
 
-            let name_flag = names.into_iter().any(|name| {
-                include_containers
-                    .iter()
-                    .any(|include| name.starts_with(include))
-            });
+        self.include_containers
+            .as_ref()
+            .map(|include_list| Self::name_or_id_matches(id, &containers, include_list))
+            .unwrap_or(false)
```

Comment:
Should this be `unwrap_or(true)`? That is, if `self.include_containers` is `None`, we'd like to default to including it.

Reply:
Ah yes. Good catch.